front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -113,4 +116,63 @@ const handleOperation = async ({
   }
 };
 
+const handleUpdateNode = debounce((timeTableId: number, node: Node) => {
+  const { betriebspunktName: trigram, positionX, positionY } = node;
+  nodeStore.set(timeTableId, { trigram, positionX, positionY });
+}, 2000);
+
+const handleNodeOperation = ({
+  type,
+  node,
+  timeTableId,
+}: {
+  type: NGEEvent['type'];
+  node: Node;
+  timeTableId: number;
+}) => {
+  switch (type) {
+    case 'create':
+    case 'update': {
+      handleUpdateNode(timeTableId, node);
+      break;
+    }
+    case 'delete': {
+      nodeStore.delete(timeTableId, node.betriebspunktName);
+      break;
+    }
+    default:
+      break;
+  }
+};
+
+const handleOperation = async ({
+  event,
+  dispatch,
+  timeTableId,
+  netzgrafikDto,
+}: {
+  event: NGEEvent;
+  dispatch: AppDispatch;
+  timeTableId: number;
+  netzgrafikDto: NetzgrafikDto;
+}) => {
+  const { type } = event;
+  switch (event.objectType) {
+    case 'node':
+      handleNodeOperation({ type, node: event.node, timeTableId });
+      break;
+    case 'trainrun':
+      await handleTrainrunOperation({
+        type,
+        trainrun: event.trainrun!,
```

Comment:
You can remove the ! here
```suggestion
        trainrun: event.trainrun,
```